src/test/mixxxtest.cpp

```diff
@@ -65,3 +68,79 @@ MixxxTest::~MixxxTest() {
         pControl->deleteCreatorCO();
     }
 }
+
+void MixxxTest::saveAndReloadConfig() {
+    m_pConfig->save();
+    m_pConfig = UserSettingsPointer(
+            new UserSettings(getTestDataDir().filePath("test.cfg")));
+    ControlDoublePrivate::setUserConfig(m_pConfig);
+}
+
+namespace test {
+
+FileRemover::~FileRemover() {
+    VERIFY_OR_DEBUG_ASSERT(
+            m_fileName.isEmpty() ||
+            QFile::remove(m_fileName) ||
+            !QFile::exists(m_fileName)) {
+        // unexpected
+    }
+}
+
+QString generateTemporaryFileName(const QString& fileNameTemplate) {
+    auto tmpFile = QTemporaryFile(fileNameTemplate);
+    // The file must be opened to create it and to obtain
+    // its file name!
```

Comment:
.. what would else happen?

Reply:
Nothing. The file is created lazily when opened.

Reply:
Please refer to the documentation of QTemporaryFile. Really tricky, it includes some platform dependent surprises.